pypesto/history/base.py

```diff
@@ -246,51 +251,117 @@ def get_trimmed_indices(self) -> np.ndarray:
         fval_trace = self.get_fval_trace()
         return np.where(fval_trace <= np.fmin.accumulate(fval_trace))[0]
 
+    def implements_trace(self) -> bool:
+        """Check whether the history has a trace that can be queried."""
+        try:
+            self.get_fval_trace()
+        except NotImplementedError:
+            return False
+
+        return True
 
-class History(HistoryBase):
+
+class NoHistory(HistoryBase):
+    """Dummy history that does not do anything.
+
+    Can be used whenever a history object is needed, but no history is desired.
+    Can be created, but not queried.
     """
-    Tracks number of function evaluations only, no trace.
 
-    Parameters
-    ----------
-    options:
-        History options.
+    def update(  # noqa: D102
```

Comment:
Those `# noqa: D102` are so ugly. Is there no way around that to not enforce docstrings in overriding methods?

Reply:
Yes, I agree. I played around with https://pypi.org/project/overrides/ and `flake8 --ignore-decorators=overrides` for a moment, but got recursion errors and decided it's not worth it ...

Reply:
or now that most functions are documented, we could ignore D102 again, and just pay attention with new prs ... would have no problem with that per se ...

Reply:
Maybe safer with D102... Okay to leave.